### PR TITLE
Initial build of Main Menu and Dropdown Submenus

### PR DESCRIPTION
**Pull request by cclen**
Summary of Code Changes:
- Build hardcoded top menu and dropdown submenus.
- Give it a style similar to the Home Depot website.
- Build JavaScript functions to ensure the expected behavior.

Dropdown hover behavior:
- Display related dropdown menu when a list item is being hovered in the main menu.
- Ensure that the dropdown remains open when it is being hovered.

Slideout hover behavior: 
- Display related slideout when a list item in the dropdown is being hovered.
- Ensure that the slideout remains open when it is being hovered.
- Ensure dropdown remains open while slideout is being hovered.

Unhover behaviors:
- When the mouse leaves all submenus, remove all previously opened submenus.
- When hover leaves slideout but hovers dropdown, display dropdown.
- When a new list item is being hovered in the dropdown, make sure to close the previous slideout.

https://user-images.githubusercontent.com/20504331/142937717-9d4ad651-2edb-4b4f-a4b7-91e7705007e2.mov

### CODE REVIEW

**Review by nbeliczay — APPROVED** [review]
Very nice!

**Comment by cclen** [comment]
This is now complete and ready for review.

File: menu-js-component/index.html [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                    <li><a id="" class="slideout-button" href="#">Lighting &amp; Ceiling Fan</a></li>
+                    <li><a id="" class="slideout-button" href="#">Outdoor Living &amp; Patio</a></li>
+                    <li><a id="" class="slideout-button" href="#">Paint</a></li>
+                    <li><a id="" class="slideout-button" href="#">Smart Home</a></li>
+                    <li><a id="" class="slideout-button" href="#">Storage &amp; Organization</a></li>
+                    <li><a id="" class="slideout-button" href="#">Tools</a></li>
+                </ul>
+            </div>
+
+            <div id="submenu-home-decor-furniture" class="submenu submenu-level-1 submenu-home-decor-furniture">
```
Comment (nbeliczay) [inline menu-js-component/index.html]: can you reuse the id instead of repeating it as a class name here?

File: menu-js-component/index.html [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    </head>
+    <body>
+        <header id="top-header">
+            <banner>
+                <p>Get Black Friday Prices Now Through December. While Supplies Last. Shop Now.</p>
+            </banner>
+
+            <!-- Main Menu - Top Menubar -->
+            <div class="main-menu-container">
+                <ul id="main-menu" class="main-menu">
```
Comment (nbeliczay) [inline menu-js-component/index.html]: use id instead of repeating it in the class name?

File: menu-js-component/index.html [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+                    <li><a href="#" id="local-ad-catalog" class="drop-button">Local Ad & Catalog</a></li>
+                </ul>
+            </div>
+
+            <!-- Level-1 Menus - Panel 1 - Dropdown -->
+
+            <div id="submenu-all-departments" class="submenu submenu-level-1 submenu-all-departments">
+                <ul>
+                    <li><a id="holiday-decorations" class="slideout-button" href="#">Holiday Decorations - subm</a></li>
+                    <li><a id="appliances" class="slideout-button" href="#">Appliances - has submenu</a></li>
```
Comment (nbeliczay) [inline menu-js-component/index.html]: is the slideout-button class necessary on these elements, or can you grab them via the parent element ?

**Review by nbeliczay — COMMENTED** [review]
left a few comments for you to review, otherwise looks good!

**Comment by cclen** [comment]
Thank you @nbeliczay for catching the unnecessary class names and approving🙏 
Merging after cleaning up the code.

**Comment by cclen** [comment]
PR #2

**Pull request merged**